src/registry-buddy/image/dynamic_deleter.go

```diff
@@ -0,0 +1,25 @@
+package image
+
+import (
+	"code.cloudfoundry.org/capi-k8s-release/src/registry-buddy/dockerhub"
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"log"
+	"strings"
+)
+
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 -o fakes/deleter.go --fake-name Deleter . Deleter
+type Deleter func(name.Reference, authn.Authenticator, *log.Logger) error
+
+func NewDynamicDeleter() Deleter {
+	return func(reference name.Reference, authenticator authn.Authenticator, logger *log.Logger) error {
+		var deleter Deleter
+		if strings.HasPrefix(reference.Context().Name(), "index.docker.io/") {
```

Comment:
Do you know if all DockerHub hosted images have this prefix?

Reply:
I'm not 100% about this, though it was the case for my deployment. It would be simple to make this a "ContainsSubstring" check instead of a prefix check. I chose the prefix check because it's the most restrictive for the case I know exists.